src/xrGame/space_restriction.cpp

```diff
@@ -152,7 +152,7 @@ CSpaceRestriction::CBaseRestrictionPtr CSpaceRestriction::merge(
     RESTRICTIONS::const_iterator I = temp_restrictions.begin();
     RESTRICTIONS::const_iterator E = temp_restrictions.end();
     for (; I != E; ++I)
-        temp = strconcat(sizeof(S), S, *temp, ",", *(*I)->name());
+        temp = strconcat(strlen(S), S, *temp, ",", *(*I)->name());
```

Comment:
`strlen` is incorrect here, because the string `S` is empty, so it would always fail. Instead `sizeof(S)` should be replaced with `acc_length`, which is the variable used as the size to allocate `S`.